### PR TITLE
Documents nullable columns in baked migrations

### PR DESCRIPTION
**Pull request by nitincoded**
This feature is being introduced in cakephp/migrations v1.6.3.

Reference: Issue cakephp/migrations#203 / PR cakephp/migrations#249

### CODE REVIEW

File: en/migrations.rst [hunk 1]
```diff
@@ -204,14 +204,16 @@ Columns definition
...
 follow the following pattern::
 
-    fieldName:fieldType[length]:indexType:indexName
+    fieldName:fieldType?[length]:indexType:indexName
 
 For instance, the following are all valid ways of specifying an email field:
 
 * ``email:string:unique``
 * ``email:string:unique:EMAIL_INDEX``
 * ``email:string[120]:unique:EMAIL_INDEX``
```
Comment (dereuromark) [inline en/migrations.rst]: maybe add an optional example here?

**Comment by dereuromark** [comment]
Does a nullable display field make sense if its marked as unique?
Maybe a description could be nullable or sth?

**Comment by nitincoded** [comment]
@dereuromark : In databases, a unique field can be nullable... at least that's the case with Oracle and MS SQL.

As an example, ReddIt permits user account signups without email addresses. However, if somebody does enter an email address, they probably would want to tell the person that the user account already exists.

**Comment by dereuromark** [comment]
Interesting, so multiple nulls are then not "unique constraint"? :) OK

File: en/migrations.rst [hunk 2]
```diff
@@ -204,14 +204,17 @@ Columns definition
 When using columns in the command line, it may be handy to remember that they
 follow the following pattern::
 
-    fieldName:fieldType[length]:indexType:indexName
+    fieldName:fieldType?[length]:indexType:indexName
 
 For instance, the following are all valid ways of specifying an email field:
 
+* ``description:string?``
```
Comment (dereuromark) [inline en/migrations.rst]: Doesn't fit to the header "For instance, the following are all valid ways of specifying an email field:"
Better to either add a new paragraph or keep the way you had it.

**Comment by nitincoded** [comment]
@dereuromark : Yes, that's an interesting case. A database column with a unique constraint allows multiple rows with nulls on Oracle, but I recall that old versions of Microsoft SQL Server had a different behavior.

File: en/migrations.rst (hunk 2 above)
Comment (nitincoded) [inline en/migrations.rst]: Shall I change the header to "For instance, the following are all valid ways of specifying a field:" ? That way, we can later provide examples with different data types too. BTW, the field type is optional and defaults to string (except for the columns id, created, modified etc.).

Comment (dereuromark) [inline en/migrations.rst]: sure :)

**Comment by dereuromark** [comment]
:+1:

File: en/migrations.rst (hunk 2 above)
Comment (nitincoded) [inline en/migrations.rst]: @dereuromark : I think I edited my comment and hit save after your reply... sorry about that

**Pull request merged**